portfolio/src/main/webapp/script.js

```diff
@@ -31,16 +31,131 @@ function addRandomGreeting() {
  * Opens a modal with the given image and caption.
  */
 function openSimpleImageModal(modalId, modalImgId, captionId, closeId, imgSrc, captionText) {  
-  var modal = document.getElementById(modalId);
-  var modalImg = document.getElementById(modalImgId);
-  var caption = document.getElementById(captionId);
+  const modal = document.getElementById(modalId);
+  const modalImg = document.getElementById(modalImgId);
+  const caption = document.getElementById(captionId);
 
-  modal.style.display = "block";
+  modal.style.display = 'block';
   modalImg.src = imgSrc;
-  caption.innerHTML = captionText;
+  caption.innerText = captionText;
 
-  var closeBtn = document.getElementById(closeId);
+  const closeBtn = document.getElementById(closeId);
   closeBtn.onclick = function() {
-    modal.style.display = "none";
+    modal.style.display = 'none';
   };
 }
+
+/**
+ * Create gallery elements and set images.
+ */
+function setGalleryImages() {
+  const slideContainer = document.getElementsByClassName('slides-container')[0];
+  const nextBtn = document.getElementsByClassName('next-slide')[0];
+  
+  // If we are on Home page gallery will not be defined.
+  if (typeof slideContainer === 'undefined') {
+    return;
+  }
+
+  for (pic of picData) {
```

Comment:
how does this work without let?

Reply:
Const is referring to the object reference, as the reference isn't changing, but the attributes are being changed this is still okay. At least that is what my knowledge of JS is, I may be mistaken

Reply:
What I meant is, I expected this line to be either `let pic of..` or `const pic of..`, but I see neither here. Your understanding of consts in JS looks correct